query-engine/query-engine-napi/src/engine.rs

```diff
@@ -292,4 +291,20 @@ impl QueryEngine {
             Inner::Builder(ref builder) => builder.logger.next_event().await,
         }
     }
+
+    /// Kicks everybody who's listening the log stream out.
+    async fn disconnect_listeners(&self) -> crate::Result<()> {
+        let inner = self.inner.read().await;
+
+        match *inner {
+            Inner::Connected(ref engine) => {
+                engine.logger.disconnect_listeners().await?;
```

Comment:
what does logger.disconnect_listeners() do? I'm too lazy to look it up

Reply:
sends a special event so javascript knows to stop listening.